### ISSUE TITLE
Accessibility: The button to enable desktop notifications should be a real button

### ISSUE THREAD

**Issue by MarcoZehe** [mozilla/discourse#233]
If one hasn't set the permission to allow desktop notifications yet, the box that asks for it has several issues. The most prominent one is that the button to enable desktop notifications is merely non-semantic text. It seems to be a custom component, but that lacks all the characteristics of the [Button design pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#button). This makes it inaccessible to both screen reader and sighted keyboard users.

* It cannot be tabbed to via the keyboard.
* It does not register to screen reader users as a button.
* Even if focused, its keyboard handler does not react to space or enter keys to perform the same action that a mouse click would initiate.

To fix, implement the above design pattern, or switch to using a regular <button> element, which gives you all of that for free.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/DISC-153)

**Comment by LeoMcA** [comment]
Created a PR to fix this upstream, by using a regular button: https://github.com/discourse/discourse/pull/8255

**Comment by LeoMcA** [comment]
This fix has been deployed on prod
